DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -2022,6 +2022,8 @@ globus_l_gfs_iRODS_recv(
 
     iRODS_handle = (globus_l_gfs_iRODS_handle_t *) user_arg;
     
+    collection = strdup(transfer_info->pathname);
```

Comment:
The `free` call for this is on L2219, but I see a few `return`s before that where this will be leaked (lines 2080, 2105, 2150, and 2161).

Reply:
done